core/src/main/java/jenkins/util/JenkinsClassLoader.java

```diff
@@ -0,0 +1,32 @@
+package jenkins.util;
+
+import jenkins.ClassLoaderReflectionToolkit;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Enumeration;
+
+/**
+ * Public versions of various {@link ClassLoader} methods for use in {@link
+ * ClassLoaderReflectionToolkit}.
+ */
+@Restricted(NoExternalUse.class)
+public interface JenkinsClassLoader {
+    /** Public version of {@link ClassLoader#findClass(String)} */
+    Class<?> findClass(String name) throws ClassNotFoundException;
+
+    /** Public version of {@link ClassLoader#findLoadedClass(String)} */
+    Class<?> findLoadedClass2(String name);
```

Comment:
Why the `2`?

Reply:
Because [`ClassLoader#findLoadedClass`](https://docs.oracle.com/javase/8/docs/api/java/lang/ClassLoader.html#findLoadedClass-java.lang.String-) is final. (Also, I am not the one who introduced `findLoadedClass2`. Tim did in a previous PR.)